src/components/GridLayout.js

```diff
@@ -28,11 +23,11 @@ class GridLayout extends Component {
     this.state = {
       width: window.innerWidth,
       height: window.innerHeight,
-      settingsMenu: false,
-      settingsMenuId: 0
+      settingsMenu: false
     };
     this.handleUpdateDimensions = this.handleUpdateDimensions.bind(this);
     this.handleKeyPress = this.handleKeyPress.bind(this);
+    this.closeSettingsMenu = this.closeSettingsMenu.bind(this);
```

Comment:
Settingsmenu is factored out to other component. The setitngsmenu itself has a close button, but the logic for showing or not showing the settings is still decided in gridlayout. Gridlayout therefore passes the closeSettingsMenu function as prop